compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -772,6 +773,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let msg = "introduce a new binding instead";
                         let sugg = format!("other_{}", ident.as_str().to_lowercase());
                         e.span_suggestion(ident.span, msg, sugg, Applicability::HasPlaceholders);
+
+                        let def_id_opt = match pat_ty.kind {
+                            Adt(def, _) => Some(def.did),
+                            _ => None,
+                        };
+                        let ranges = &[
+                            self.tcx.lang_items().range_struct(),
+                            self.tcx.lang_items().range_from_struct(),
+                            self.tcx.lang_items().range_to_struct(),
+                            self.tcx.lang_items().range_full_struct(),
+                            self.tcx.lang_items().range_inclusive_struct(),
+                            self.tcx.lang_items().range_to_inclusive_struct(),
+                        ];
+                        if ranges.contains(&def_id_opt) {
```

Comment:
This will search for a `None` lang item when `def_id_opt` is `None`, which is probably not right. The note should only be emitted when `pat_ty.kind` is `Adt`.

Reply:
my understanding is that the .lang_items().range_* above will never be none, but you are right, an explicit != None check would be good

Reply:
`.lang_items().range_*` might be `None` is special setups (I think `no_std` might be one of them).